### ISSUE TITLE
net: net.OpError.Temporary different between windows and linux.

### ISSUE THREAD

**Issue by eahydra** [golang/go#15637]
What version of Go are you using (`go version`)?
go version go1.6 windows/amd64

I write a Redis Client with the net package. And I want to check error except timeout error such as Connection Reset or Connection aborted. 

I read src/syscall/syscall_windows.go and find Errno.Temporary() just check EINTR, EMFILE and Errno.Timeout, but the src/syscall/syscall_unix.go, Errno.Temporary() additional check ECONNRESET and ECONNABORTED.

so why different between windows and linux?

Sorry my poor English.

**Comment by alexbrainman** [comment]
> why different between windows and linux?

Originally Go had linux and darwain, but no support for windows. So when adding windows, we had to define all unix constants like EINTR and EMFILE, because os package used to use them. Otherwise os package wouldn't build.

But Windows syscalls do not really return EINTR or EMFILE or ECONNRESET or ECONNABORTED.

When time came to implement net package, windows version was very much separate code. So net package does not use ECONNRESET or ECONNABORTED.

I just checked our current source:
```
# cd $GOROOT
# grep syscall.ECONNRESET * -r
# grep syscall.ECONNABORTED * -r
src/net/fd_unix.go:                     case syscall.ECONNABORTED:
src/net/error_unix_test.go:     abortedConnRequestErrors = []error{syscall.ECONNABORTED} // see accept in fd_unix.go
#
```

No windows code uses ECONNRESET or ECONNABORTED.

Alex

**Comment by eahydra** [comment]
Yes. windows WIN32 API do not really return ECONNRESET,just return such as WSAECONNRESET.
Hmm, I don't known whether if it is a bug. When I want to check net.OpError to determine CONNREST like this: https://play.golang.org/p/hnxswTG_Sl (forgive me.. I also want to markdown to paste the code, but the format....)

So, Is the way right? Or is there another way to do this?

**Comment by alexbrainman** [comment]
> So, Is the way right? Or is there another way to do this?

I don't know. I will let @mikioh and @bradfitz answer this.

Alex

**Comment by rsc** [comment]
Basically none of the actual temporary/timeout errors are checked by Errno.Temporary and Errno.Timeout. Like you said, all the WSA* forms are missing to start. I am not sure how to enumerate all the ones that should match, though. There are many more error codes on Windows than on Unix.

**Comment by MaxSem** [comment]
After 7f6105f13, there's no more discrepancy between Unix and Windows. Is adding Windows-specific error codes still desireable?
